sources/packages/api/src/services/application/application.service.ts

```diff
@@ -638,4 +646,129 @@ export class ApplicationService extends RecordDataModelService<Application> {
+      throw new CustomNamedError(
+        "Student Application not found or the location does not have access to it",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (
+      appToOverride.applicationStatus !== ApplicationStatus.enrollment ||
+      appToOverride.coeStatus !== COEStatus.required
+    ) {
+      throw new CustomNamedError(
+        `Student Application is not in the expected status. The application must be in application status '${ApplicationStatus.enrollment}' and COE status '${COEStatus.required}' to be override.`,
+        INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+      );
+    }
+
+    // Change status of the current application being overwritten.
+    appToOverride.applicationStatus = ApplicationStatus.overwritten;
+    appToOverride.applicationStatusUpdatedOn = getUTCNow();
+
+    // Create a new application based in the previous one.
+    const newApplication = new Application();
+    newApplication.data = appToOverride.data;
+    newApplication.applicationNumber = appToOverride.applicationNumber;
+    newApplication.student = {
+      id: appToOverride.studentId,
+    } as Student;
+    newApplication.location = {
+      id: appToOverride.locationId,
+    } as InstitutionLocation;
+    if (appToOverride.pirProgramId) {
+      newApplication.pirProgram = {
+        id: appToOverride.pirProgramId,
+      } as EducationProgram;
+    }
+    if (appToOverride.offeringId) {
+      newApplication.offering = {
+        id: appToOverride.offeringId,
+      } as EducationProgramOffering;
+    }
+    newApplication.programYear = {
+      id: appToOverride.programYearId,
+    } as ProgramYear;
+    newApplication.pirStatus = ProgramInfoStatus.required;
+    newApplication.applicationStatus = ApplicationStatus.submitted;
+    newApplication.applicationStatusUpdatedOn = getUTCNow();
+    newApplication.studentFiles = appToOverride.studentFiles.map(
+      (applicationFile: ApplicationStudentFile) => {
+        // Recreate file associations for new application.
+        const fileAssociation = new ApplicationStudentFile();
+        fileAssociation.studentFile = {
+          id: applicationFile.studentFileId,
+        } as StudentFile;
+        return fileAssociation;
+      },
+    );
+
+    await this.repo.save([appToOverride, newApplication]);
+
+    return {
+      overriddenApplication: appToOverride,
+      createdApplication: newApplication,
+    };
+  }
+
+  /**
+   * Starts an application assessment workflow associating
+   * the created workflow with the Student Application.
+   * @param applicationId Student Application to have the
+   * workflow started.
+   */
+  async startApplicationAssessment(applicationId: number) {
+    const application = await this.repo.findOneOrFail(applicationId);
+    if (!application) {
+      throw new CustomNamedError(
+        "Student Application not found.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (application.applicationStatus !== ApplicationStatus.submitted) {
```

Comment:
got a suggestion in my last PR to use `Yoda notation` from @guru-aot . May be we can follow that in our application. just suggestion

Reply:
I don't mind if we have a proper justification. Why it would be a benefit to use it in typescript?